### PR TITLE
Clean up all files using the pre-commit hook

### PR DESCRIPTION
**Pull request by kayman-mk**
### Brief summary of the change made
Last week the  pre-commit hook was introduced. By default it runs for changed files in the feature branch. But we have to sanitize the main branch as well.

I ran the pre-commit hook and committed all changes.

### Pull Request checklist
- [x] Please confirm you have completed any of the necessary steps below.

- Included test cases to demonstrate any code changes, which may be one or more of the following:
  - `.yml` rule test cases in `test/fixtures/rules/std_rule_cases`.
  - `.sql`/`.yml` parser test cases in `test/fixtures/dialects` (note YML files can be auto generated with `python test/generate_parse_fixture_yml.py` or by running `tox` locally).
  - Full autofix test cases in `test/fixtures/linter/autofix`.
  - Other.
- Added appropriate documentation for the change.
- Created GitHub issues for any relevant followup/future enhancements if appropriate.

### CODE REVIEW

**Comment by kayman-mk** [comment]
@jpy-git I cleaned all files using the pre-commit hook

**Comment by kayman-mk** [comment]
Oh, not an easy one. The pipeline fails with an SqlFluff error now.

And there is one problem reported from the pre-commit hook. Not sure how to deal with it:
```
black....................................................................Passed
mypy.....................................................................Failed
- hook id: mypy
- exit code: 2

test/__init__.py: error: Duplicate module named "test" (also at "plugins/sqlfluff-templater-dbt/test/__init__.py")
test/__init__.py: note: Are you missing an __init__.py? Alternatively, consider using --exclude to avoid checking one of them.
Found 1 error in 1 file (errors prevented further checking)
```

**Comment by jpy-git** [comment]
@kayman-mk so the reason the tests are failing now is because the pre-commit is affecting .sql files. However, there are genuine unit tests that have missing trailing newlines and excess trailing whitespace as these are things the linter aims to fix.

So what we need to do is update the top of `.pre-commit-config.yaml` to exclude sql files as we don't want to mess with their formatting:
```
-   repo: https://github.com/pre-commit/pre-commit-hooks
    rev: v4.0.1
    hooks:
    -   id: end-of-file-fixer
        exclude: .*\.sql$
    -   id: trailing-whitespace
        exclude: .*\.sql$
```

**Comment by jpy-git** [comment]
and for mypy we only care about the src folder so change that to:
```
-   repo: https://github.com/pre-commit/mirrors-mypy
    rev: v0.910-1
    hooks:
    -   id: mypy
        args: [--ignore-missing-imports]
        additional_dependencies: [
            types-toml,
            types-pkg_resources,
            types-chardet,
        ]
        files: ^src/sqlfluff/.*
```

**Comment by kayman-mk** [comment]
Waiting for #2125

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -3,7 +3,9 @@ repos:
     rev: v4.0.1
     hooks:
     -   id: end-of-file-fixer
+        exclude: "test/fixtures/linter/.*.sql|plugins/sqlfluff-templater-dbt/test/fixtures/dbt/dbt_project/models/.*.sql|plugins/sqlfluff-templater-dbt/test/fixtures/dbt/templated_output/.*.sql|test/fixtures/templater/.*.sql"
     -   id: trailing-whitespace
+        exclude: "test/fixtures/linter/.*.sql|plugins/sqlfluff-templater-dbt/test/fixtures/dbt/dbt_project/models/.*.sql|plugins/sqlfluff-templater-dbt/test/fixtures/dbt/templated_output/.*.sql|test/fixtures/templater/.*.sql"
```
Comment (jpy-git) [inline .pre-commit-config.yaml:8]:
```suggestion
    -   id: end-of-file-fixer
        exclude: .*\.sql$
    -   id: trailing-whitespace
        exclude: .*\.sql$
```
Not sure why you didn't just use the regex I gave you

**Review by jpy-git — CHANGES_REQUESTED** [review]
@kayman-mk you need to remove all the sql files that got added to this PR

**Comment by kayman-mk** [comment]
Thought it's better not to include every sql file but only the relevant ones.

**Comment by kayman-mk** [comment]
@jpy-git ok, one failing check now. Something with the windows system. No idea what's going wrong. Do you?

**Comment by jpy-git** [comment]
@kayman-mk no worries, we're likely to add more test sql in other locations in the future so safest to just not mess with any of their formatting, looks good now so thanks 😄 

I can see that in your last commit you've indented the whole `.pre-commit-config.yaml` file which is messing up the diff, could you dedent that so we only see the lines that you've changed please?

**Comment by jpy-git** [comment]
ah interesting let me take a look at that windows test 🤔

File: test/fixtures/linter/sqlfluffignore/path_c [hunk 2]
```diff
@@ -1 +1 @@
-.
\ No newline at end of file
+.
```
Comment (jpy-git) [inline test/fixtures/linter/sqlfluffignore/path_c:1]: @kayman-mk this seems to be the change that's upsetting windows, investigating further

Comment (jpy-git) [reply test/fixtures/linter/sqlfluffignore/path_c:1]: lol why is there an infinitely repeating symlink here
image

Comment (jpy-git) [reply test/fixtures/linter/sqlfluffignore/path_c:1]: Interestingly when I ran `tox -e pre-commit -- run --all-files` I had all the same files get fixed except this path_c symlink:
image

Comment (jpy-git) [reply test/fixtures/linter/sqlfluffignore/path_c:1]: @kayman-mk out of curiosity which OS are you using and did you run the same command as me?

I think to be safe I'll add this file to the exludes in pre-commit-config

**Review by jpy-git — COMMENTED** [review]
Could you also reset test/fixtures/linter/sqlfluffignore/path_c to be unchanged by this PR + dedent `pre-commit-config.yaml` as mentioned in previous comment 👍 

Some weird symlink test going on that we shouldn't mess with. 🤷

File: test/fixtures/linter/sqlfluffignore/path_c (hunk 2 above)
Comment (kayman-mk) [reply test/fixtures/linter/sqlfluffignore/path_c:1]: No idea what's going on here. Using Windows with IntelliJ, but no Python installed. So I do the pre-commit in a Alpine docker image.

**Comment by kayman-mk** [comment]
`pre-commit-config.yaml`: I remember a merge conflict with the main branch, but I am pretty sure that I didn't change the indentation. Anyway, fixed now.

**Review by jpy-git — APPROVED** [review]
Looks good, thanks for working through this ⭐ LGTM 🚀

**Pull request merged**